### PR TITLE
633176: Ensure staging queue is declared when routing to staging queue

### PR DESCRIPTION
**Pull request by rorytorneymf**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=633176

Moving staging queue creation to before where `queueDeclarePassive` is called. Previously, `queueDeclarePassive` was being called on the original target queue rather than the staging queue. When a staging queue is deleted now, an error will be logged and the job retried, rather than the message being lost by being sent to a deleted queue.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -93,7 +93,7 @@
             <dependency>
                 <groupId>com.github.cafapi</groupId>
                 <artifactId>caf-dependency-management-bom</artifactId>
-                <version>3.4.0-SNAPSHOT</version>
+                <version>3.4.0-633176-SNAPSHOT</version>
```
Comment (rorytorneymf) [inline pom.xml]: Remove dev branch before merging

File: pom.xml [hunk 2]
```diff
@@ -210,7 +210,7 @@
             <dependency>
                 <groupId>com.github.workerframework</groupId>
                 <artifactId>worker-message-prioritization-rerouting</artifactId>
-                <version>1.0.0-SNAPSHOT</version>
+                <version>1.0.0-633176-SNAPSHOT</version>
```
Comment (rorytorneymf) [inline pom.xml]: Remove dev branch before merging

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/ExceptionThrowingReturnListener.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                             final String exchange,
+                             final String routingKey,
+                             final AMQP.BasicProperties properties,
+                             final byte[] body)
+    {
+        final String errorMessage = String.format(
+                "Message returned from RabbitMQ: replyCode=%d, replyText=%s, exchange=%s, routingKey=%s, properties=%s, body=%s",
+                replyCode, replyText, exchange, routingKey, properties, new String(body));
+
+        throw new RuntimeException(errorMessage);
```
Comment (andyreidz) [inline job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/ExceptionThrowingReturnListener.java]: Can you describe the behaviour here? What is catching this exception? When it's thrown, what happens?

**Review by andyreidz — CHANGES_REQUESTED** [review]
As discussed, let's remove ReturnListener

File: job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/DatabasePoller.java [hunk 4]
```diff
@@ -73,8 +73,7 @@ public static void pollDatabaseForJobsToRun() {
 
     private static void sendMessageToQueueMessaging(final Codec codec, final JobTaskData jtd, final WorkerAction workerAction)
     {
-        try (final QueueServices queueServices= QueueServicesFactory.create(jtd.getTaskPipe(), codec)){
-            LOG.debug("Sending task data to the target queue {} ...", workerAction);
```
Comment (rorytorneymf) [inline job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/DatabasePoller.java:77]: I removed this debug log as its essentially a duplicate of this log:

https://github.com/JobService/job-service/pull/332/files#diff-71727cc500796b7649ac435047b76c4b12f456408f532e53a037918ec9d6bdc6R119

**Review by andyreidz — APPROVED** [review]

**Pull request merged**